lib/northbound.c

```diff
@@ -142,18 +139,18 @@ static int nb_node_new_cb(const struct lys_node *snode, void *arg)
 	 */
 	nb_node->snode = snode;
 	assert(snode->priv == NULL);
-	lys_set_private(snode, nb_node);
+	((struct lysc_node *)snode)->priv = nb_node;
```

Comment:
Yes, I disliked this too. However, the issue is that it really fans out and you have to duplicate a bunch of code b/c your changing the callback function type. So you need 2 callbacks a mutable one and a const one, then you need to duplicate your iterate functions so that they take a mutable callback or a const one, etc.. all just for this one instance where we actually change the snode. It's one of those calls you have to make if being super "clean" is worth actually adding a ton of extra distracting code, or if just adding a comment to note the singular deviation is better (I think the latter is true here).